### PR TITLE
Adding ability to specify branch or commit to use for the community n…

### PR DESCRIPTION
**Pull request by josmo**
…etworks.

Helps to be able to specify a commit or branch to groups that are doing community managed builds

### CODE REVIEW

**Comment by nicopace** [comment]
@josmo could you add information on how would you use it on the README file?

That way would be easy both to discuss about the functionality, and to test the implementation.

Thanks!

**Comment by josmo** [comment]
@nicopace Let me know if that works? :)

File: cooker [hunk 1]
```diff
@@ -297,12 +297,23 @@ download_feeds() {
 }
 
 download_communities() {
-    [ ! -d "$communities_dir" ] && {
+    [ ! -d "$communities_git" ] && {
         echo "-> Downloading communities repository"
-        git clone $communities_git $communities_dir || {
+        if echo "$communities_git" | grep \;; then
+            url="$(echo $communities_git | awk -F\; '{print $1}')"
```
Comment (aparcar) [inline cooker:303]: does `${communities_git##*;}` works for you?

File: cooker [hunk 2]
```diff
@@ -297,12 +297,23 @@ download_feeds() {
...
+            branch="$(echo $communities_git | awk -F\; '{print $2}')"
```
Comment (aparcar) [inline cooker:304]: `${communities_git%%;*}`

File: cooker [hunk 3]
```diff
@@ -297,12 +297,23 @@ download_feeds() {
...
+            git clone $url -b $branch $communities_dir
+        elif echo "$communities_git" | grep \^; then
+            url="$(echo $communities_git | awk -F\^ '{print $1}')"
```
Comment (aparcar) [inline cooker:307]: same here and below

File: cooker [hunk 4]
```diff
@@ -297,12 +297,23 @@ download_feeds() {
...
+            commit="$(echo $communities_git | awk -F\^ '{print $2}')"
+            git clone $url $communities_dir
+            ( cd $communities_dir && git checkout $commit )
+        else
+            git clone $communities_git $communities_dir || {
```
Comment (aparcar) [inline cooker:312]: always add `"` around vars

File: options.conf [hunk 5]
```diff
@@ -1,6 +1,6 @@
 release=17.01.4
 base_url=https://downloads.lede-project.org/releases/$release/targets/
-communities_git=https://github.com/libremesh/network-profiles.git
+communities_git='https://github.com/libremesh/network-profiles.git'
```
Comment (aparcar) [inline options.conf:3]: I'd leave this as is and only add braces if needed - but just my opinion

**Review by aparcar — CHANGES_REQUESTED** [review]

**Comment by nicopace** [comment]
Love the idea!
Thanks @aparcar for the review.
@josmo if you go through aparcar's corrections, it is good to go for me :)

**Pull request closed without merging**